File: leetcode/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0, buyPrice = prices[0];
        
        for (int i = 1; i < prices.size(); i++) {
            if (maxProfit < prices[i] - buyPrice)
                maxProfit = prices[i] - buyPrice;
            
            if (prices[i] < buyPrice)
                buyPrice = prices[i];
        }
        
        return maxProfit;
    }
};

File: leetcode/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        buyPrice = prices[0]
        
        for i in range(1, len(prices)):
            if maxProfit < prices[i] - buyPrice:
                maxProfit = prices[i] - buyPrice
            
            if prices[i] < buyPrice:
                buyPrice = prices[i]
        
        return maxProfit;